Not compiling on CC7 with clang 3.9

Hi,
using clang 3.9 with  libstdc++ form gcc 6.2 it does not compile

```
source /cvmfs/clicdp.cern.ch/compilers/llvm/3.9.0/x86_64-centos7/setup.sh
source /cvmfs/clicdp.cern.ch/software/CMake/3.6.2/x86_64-centos7-llvm39-opt/setup.sh 
source /cvmfs/clicdp.cern.ch/software/Python/2.7.12/x86_64-centos7-llvm39-opt/setup.sh 
```

`cmake -DCMAKE_INSTALL_PREFIX=/some_path ../SAS-master/`

```
In file included from /sas/SAS-master/src/CodingConventions/General/StdPrintoutsChecker.h:7:
/sas/SAS-master/src/SasChecker.h:98:23: error: no viable conversion from 'clang::ento::BugReport *' to 'std::unique_ptr<BugReport>'
         C.emitReport(br);
                      ^~
/cvmfs/clicdp.cern.ch/compilers/gcc/6.2.0/x86_64-centos7/lib/gcc/x86_64-pc-linux-gnu/6.2.0/../../../../include/c++/6.2.0/bits/unique_ptr.h:201:17: note: candidate constructor not viable: no known conversion from 'clang::ento::BugReport *' to 'nullptr_t' for
      1st argument
      constexpr unique_ptr(nullptr_t) noexcept : unique_ptr() { }
                ^
/cvmfs/clicdp.cern.ch/compilers/gcc/6.2.0/x86_64-centos7/lib/gcc/x86_64-pc-linux-gnu/6.2.0/../../../../include/c++/6.2.0/bits/unique_ptr.h:206:7: note: candidate constructor not viable: no known conversion from 'clang::ento::BugReport *' to
      'std::unique_ptr<clang::ento::BugReport, std::default_delete<clang::ento::BugReport> > &&' for 1st argument
      unique_ptr(unique_ptr&& __u) noexcept
      ^
/cvmfs/clicdp.cern.ch/compilers/gcc/6.2.0/x86_64-centos7/lib/gcc/x86_64-pc-linux-gnu/6.2.0/../../../../include/c++/6.2.0/bits/unique_ptr.h:356:7: note: candidate constructor not viable: no known conversion from 'clang::ento::BugReport *' to 'const
      std::unique_ptr<clang::ento::BugReport, std::default_delete<clang::ento::BugReport> > &' for 1st argument
      unique_ptr(const unique_ptr&) = delete;
      ^
/cvmfs/clicdp.cern.ch/compilers/gcc/6.2.0/x86_64-centos7/lib/gcc/x86_64-pc-linux-gnu/6.2.0/../../../../include/c++/6.2.0/bits/unique_ptr.h:220:2: note: candidate template ignored: could not match 'unique_ptr<type-parameter-0-0, type-parameter-0-1>' against
      'clang::ento::BugReport *'
        unique_ptr(unique_ptr<_Up, _Ep>&& __u) noexcept
        ^
/cvmfs/clicdp.cern.ch/compilers/gcc/6.2.0/x86_64-centos7/lib/gcc/x86_64-pc-linux-gnu/6.2.0/../../../../include/c++/6.2.0/bits/unique_ptr.h:228:2: note: candidate template ignored: could not match 'auto_ptr<type-parameter-0-0>' against
      'clang::ento::BugReport *'
        unique_ptr(auto_ptr<_Up>&& __u) noexcept;
        ^
/cvmfs/clicdp.cern.ch/compilers/llvm/3.9.0/x86_64-centos7/include/clang/StaticAnalyzer/Core/PathSensitive/CheckerContext.h:261:46: note: passing argument to parameter 'R' here
  void emitReport(std::unique_ptr<BugReport> R) {
                                             ^
13 warnings and 1 error generated.
make[2]: *** [CMakeFiles/libSas.dir/src/CodingConventions/General/StdPrintoutsChecker.cpp.o] Error 1
make[1]: *** [CMakeFiles/libSas.dir/all] Error 2
```

--- a/src/IncludeDumper.h
+++ b/src/IncludeDumper.h
@@ -49,7 +49,7 @@ class InclusionDump : public clang::ASTConsumer {
 class InclusionDumpAction : public clang::PluginASTAction {
   public:
     InclusionDumpAction() : m_dirMask("") {}
-    bool ParseArgs(const clang::CompilerInstance &CI, const std::vector<std::string>& args);
+    virtual bool ParseArgs(const clang::CompilerInstance &CI, const std::vector<std::string>& args) override;
 
   protected:
     std::unique_ptr<clang::ASTConsumer> CreateASTConsumer(clang::CompilerInstance &CI,

--- a/src/Performance/FiniteMathChecker.cpp
+++ b/src/Performance/FiniteMathChecker.cpp
@@ -21,7 +21,7 @@ namespace sas {
              !II->isStr("__isnan") && !II->isStr("__isinf") &&
              !II->isStr("__isnanf") && !II->isStr("__isinff")) return;
 
-         clang::ento::ExplodedNode *N = C.generateSink();
+         clang::ento::ExplodedNode *N = C.generateErrorNode();
          if (!N) return;
 
          const char* msg = "[sas.Performance.FiniteMathChecker] The function isnan/isinf does not work when fast-math is enabled. Please check the bits of the floating point number instead.";

--- a/src/SasChecker.h
+++ b/src/SasChecker.h
@@ -93,9 +93,9 @@ public:
       if(bw.isDisabled()) return;
       if (auto errorNode = C.addTransition()) {
          auto bt = new clang::ento::BugType(this, Traits::BugName, Traits::BugCategory);
-         auto br = new clang::ento::BugReport(*bt, msg, errorNode);
+         auto br = llvm::make_unique<clang::ento::BugReport>(*bt, msg, errorNode);
          br->addRange(E->getSourceRange());
-         C.emitReport(br);
+         C.emitReport(std::move(br));
       }
    }
 };

--- a/src/jsonxx.h
+++ b/src/jsonxx.h
@@ -200,24 +200,24 @@ class Value {
     type_ = BOOL_;
     bool_value_ = b;
   }
-#define $number(TYPE) \
+#define _number(TYPE) \
   void import( const TYPE &n ) { \
     reset(); \
     type_ = NUMBER_; \
     number_value_ = static_cast<long double>(n); \
   }
-  $number( char )
-  $number( int )
-  $number( long )
-  $number( long long )
-  $number( unsigned char )
-  $number( unsigned int )
-  $number( unsigned long )
-  $number( unsigned long long )
-  $number( float )
-  $number( double )
-  $number( long double )
-#undef $number
+  _number( char )
+  _number( int )
+  _number( long )
+  _number( long long )
+  _number( unsigned char )
+  _number( unsigned int )
+  _number( unsigned long )
+  _number( unsigned long long )
+  _number( float )
+  _number( double )
+  _number( long double )
+#undef _number
 #if JSONXX_COMPILER_HAS_CXX11 > 0
   void import( const std::nullptr_t & ) {
     reset();
@@ -382,7 +382,7 @@ const T& Object::get(const std::string& key, const typename identity<T>::type& d
     return default_value;
   }
 }
-    
+
 template<>
 inline bool Value::is<Value>() const {
     return true;
@@ -417,12 +417,12 @@ template<>
 inline bool Value::is<Object>() const {
   return type_ == OBJECT_;
 }
-    
+
 template<>
 inline Value& Value::get<Value>() {
     return *this;
 }
-    
+
 template<>
 inline const Value& Value::get<Value>() const {
     return *this;

